rules/consistent-function-scoping.js

```diff
@@ -121,30 +121,48 @@ const create = context => {
 	const sourceCode = context.getSourceCode();
 	const {scopeManager} = sourceCode;
 
+	const reports = [];
+	let hasJsx = false;
+
 	return {
-		ArrowFunctionExpression(node) {
+		ArrowFunctionExpression: node => {
 			const valid = checkNode(node, scopeManager);
 
 			if (valid) {
-				return;
+				reports.push(null);
+			} else {
+				reports.push({
+					node,
+					messageId: MESSAGE_ID_ARROW
+				});
 			}
-
-			context.report({
-				node,
-				messageId: MESSAGE_ID_ARROW
-			});
 		},
-		FunctionDeclaration(node) {
+		FunctionDeclaration: node => {
 			const valid = checkNode(node, scopeManager);
 
 			if (valid) {
-				return;
+				reports.push(null);
+			} else {
+				reports.push({
+					node,
+					messageId: MESSAGE_ID_FUNCTION
+				});
+			}
+		},
+		JSXElement: () => {
+			// Turn off this rule if we see a JSX element because scope
+			// references does not include JSXElement nodes.
```

Comment:
Copying [comment](https://github.com/MrHen/eslint-plugin-unicorn/pull/1#discussion_r325481128) from @sindresorhus:

> Can you open an ESLint issue about this afterwards?

Looks like there was a similar issue opened here: https://github.com/eslint/eslint/issues/3622

I can try getting something like that hooked up. Ideally, if someone who uses `eslint-plugin-unicorn` also had `eslint-plugin-react` then it would "just work". I'm not sure if we can realistically test for that, though.